ansible_base/api_documentation/postprocessing_hooks.py

```diff
@@ -0,0 +1,466 @@
+
+    if placeholder_index == -1:
+        return None
+
+    # Check if there are resource segments after the placeholder
+    if not _has_segments_after_placeholder(parts, placeholder_index):
+        return None  # Not truly nested
+
+    # Extract and return parent resource name
+    return _extract_parent_from_prefix(parts, placeholder_index)
+
+
+def extract_action_and_resource(operation_id, path):
+    """
+    Extract action, resource parts, and parent resource from operation_id and path.
+    Returns: (action, resource_parts, parent_resource)
+    """
+    # Extract action using utility function
+    action = extract_operation_action(operation_id)
+
+    # Extract resource parts from operation_id
+    resource_parts = _extract_resource_parts_from_operation_id(operation_id, action)
+
+    # Detect parent-child relationships from path
+    parent_resource = _find_parent_resource_from_path(path)
+
+    # Fallback to extracting from path if no resource_parts
+    if not resource_parts:
+        path_parts = parse_path_segments(path)
+        resource_parts = [path_parts[-1]] if path_parts else ['resource']
+
+    return action, resource_parts, parent_resource
+
+
+def format_compound_resource(resource_parts, parent_resource, action):
+    """
+    Format compound resource names with proper prepositions for nested resources.
+    E.g., "users for a team", "route for an HTTP port"
+    """
+    resource_name = ' '.join(resource_parts).replace('_', ' ')
+
+    # Handle nested resources with generic "for" preposition
+    if len(resource_parts) >= 2 and parent_resource:
+        child_resource = resource_parts[-1]
+
+        # Determine article (a/an) based on parent resource
+        article = 'an' if parent_resource[0].lower() in 'aeiou' else 'a'
+
+        # For list actions, use plural child
+        if action == 'list':
+            return f"{child_resource} for {article} {parent_resource}"
+        else:
+            # For singular actions, singularize child
+            singular_child = singularize_resource(child_resource)
+            return f"{singular_child} for {article} {parent_resource}"
+
+    return resource_name
+
+
+def generate_description_from_purpose(resource_purpose, action):
+    """Generate x-ai-description from resource_purpose using action templates."""
+    # Templates optimized for MCP tool understanding
+    action_templates = {
+        'list': lambda p: f"List {p}",
+        'retrieve': lambda p: f"Retrieve a {singularize_resource_purpose(p)}",
+        'read': lambda p: f"Retrieve a {singularize_resource_purpose(p)}",
+        'create': lambda p: f"Create a {singularize_resource_purpose(p)}",
+        'update': lambda p: f"Update a {singularize_resource_purpose(p)}",
+        'partial_update': lambda p: f"Update a {singularize_resource_purpose(p)}",
+        'destroy': lambda p: f"Delete a {singularize_resource_purpose(p)}",
+        'delete': lambda p: f"Delete a {singularize_resource_purpose(p)}",
+    }
```

Comment:
Should this be a global like `OPERATION_DESCRIPTIVE_NAMES` or even merged with it?